control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -535,6 +587,111 @@ func (r *HostedControlPlaneReconciler) ensureControlPlane(ctx context.Context, h
+		return ReconcileEtcdPeerSecret(peerSecret, rootCASecret)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd peer secret: %w", err)
+	}
+
+	// Etcd Operator ServiceAccount
+	operatorServiceAccount := manifests.EtcdOperatorServiceAccount(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operatorServiceAccount), operatorServiceAccount); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get etcd operator service account: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, operatorServiceAccount, func() error {
+		operatorServiceAccount.OwnerReferences = ensureHCPOwnerRef(hcp, operatorServiceAccount.OwnerReferences)
+		return nil
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd operator service account: %w", err)
+	}
+
+	// Etcd operator role
+	operatorRole := manifests.EtcdOperatorRole(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operatorRole), operatorRole); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get etcd operator role: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, operatorRole, func() error {
+		operatorRole.OwnerReferences = ensureHCPOwnerRef(hcp, operatorRole.OwnerReferences)
+		return ReconcileEtcdOperatorRole(operatorRole)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd operator role: %w", err)
+	}
+
+	// Etcd operator rolebinding
+	operatorRoleBinding := manifests.EtcdOperatorRoleBinding(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operatorRoleBinding), operatorRoleBinding); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get etcd operator role binding: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, operatorRoleBinding, func() error {
+		operatorRoleBinding.OwnerReferences = ensureHCPOwnerRef(hcp, operatorRoleBinding.OwnerReferences)
+		return ReconcileEtcdOperatorRoleBinding(operatorRoleBinding)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd operator role binding: %w", err)
+	}
+
+	// Etcd operator deployment
+	operatorDeployment := manifests.EtcdOperatorDeployment(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operatorDeployment), operatorDeployment); err != nil && !apierrors.IsNotFound(err) {
+		return fmt.Errorf("cannot get etcd operator deployment: %w", err)
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, r, operatorDeployment, func() error {
+		operatorDeployment.OwnerReferences = ensureHCPOwnerRef(hcp, operatorDeployment.OwnerReferences)
+		return ReconcileEtcdOperatorDeployment(operatorDeployment)
+	}); err != nil {
+		return fmt.Errorf("failed to reconcile etcd operator deployment: %w", err)
+	}
+
+	// Etcd cluster
+	etcdCluster := manifests.EtcdCluster(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(etcdCluster), etcdCluster); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
why do we need this Get request here? what is it giving us that is not already handled by CreateOrUpdate?